task 15 commit1

renvema opened the pull request:
Have a nice day. )

my scanner don't work well. Can you help me please.

Commit:
task 15 commit1

Sofasmile commented on src/dao/Main.java:
```diff
@@ -0,0 +1,42 @@
+        System.out.println(userDao.getAll());
+
+    }
+
```
remove all redundant lines

Sofasmile commented on src/dao/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,55 @@
+public class ConsoleHandler {
+    private static Scanner scanner = new Scanner(System.in);
+    @Inject
+    private static BetDao betDao;
```
better add a blank line before annotation

Sofasmile commented on src/dao/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,55 @@
+    @Inject
+    private static UserDao userDao;
+
+    static public void handle() {
```
```suggestion
    public static void handle() {
```

Sofasmile commented on src/dao/controller/ConsoleHandler.java:
```diff
@@ -0,0 +1,55 @@
+        }
+    }
+
+    static public void handleUser() {
```
```suggestion
    public static void handleUser() {
```

Sofasmile commented on src/dao/controller/UserHandler.java:
```diff
@@ -0,0 +1,31 @@
+//package dao.controller;
```
do we need this file here?

Sofasmile commented on src/dao/dao/BetDao.java:
```diff
@@ -0,0 +1,15 @@
+
+    void add(Bet bet);
+    List<Bet> getAll();
+
```
redundant lines

Sofasmile commented on src/dao/dao/UserDao.java:
```diff
@@ -0,0 +1,10 @@
+
+import java.util.List;
+
+public interface UserDao {
```
optional: we can use one interface for two classes (BetDao, UserDao)

Sofasmile commented on src/dao/dao/UserDaoImpl.java:
```diff
@@ -0,0 +1,21 @@
+    public List<User> getAll() {
+        return Storage.user;
+    }
+}
```
add a blank line after }

Sofasmile commented on src/dao/db/Storage.java:
```diff
@@ -0,0 +1,13 @@
+
+//ЭТО НАША БД
+public class Storage {
+    public static final List<Bet> bets = new ArrayList<>();
```
```suggestion
    public static final List<Bet> BETS = new ArrayList<>();
```

Sofasmile commented on src/dao/db/Storage.java:
```diff
@@ -0,0 +1,13 @@
+//ЭТО НАША БД
+public class Storage {
+    public static final List<Bet> bets = new ArrayList<>();
+    public static final List<User> user = new ArrayList<>();
```
```suggestion
    public static final List<User> USERS = new ArrayList<>();
```

Sofasmile commented on src/dao/lib/Injector.java:
```diff
@@ -0,0 +1,30 @@
+
+        Field[] consoleHandlerFields = consoleHandlerClass.getDeclaredFields();
+        for (Field field : consoleHandlerFields) {
+            if (field.getDeclaredAnnotation(Inject.class) != null) {
```
add checking for annotation @ Dao

Commit:
task 15 commit2
-remove all redundant lines;
-add a blank line before annotation;
-add checking for annotation @ Dao;

Sofasmile commented on src/dao/lib/Injector.java:
```diff
@@ -0,0 +1,31 @@
+        for (Field field : consoleHandlerFields) {
+            if (field.getDeclaredAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(betDaoImplClass)) {
```
add:
betDaoImplClass.getAnnotation(Dao.class)

Sofasmile reviewed (commented):
sorry I didn't see this: my scanner don't work well
Did you fix it?

Commit:
task 15 commit3
-add betDaoImplClass.getAnnotation(Dao.class);
- scanner worked yesterday< when I push commit 2
- now, after adding betDaoImplClass.getAnnotation(Dao.class), it again sin't working. ((

Have a nice day)

Commit:
task 15 commit4
-add UserHandler.java, because forgot add it last time

Commit:
task 15 commit4
-scanner is working;

Sofasmile commented on src/dao/lib/Injector.java:
```diff
@@ -0,0 +1,46 @@
+        Field[] consoleHandlerFields = consoleHandlerClass.getDeclaredFields();
+        for (Field field : consoleHandlerFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                if (field.getType().equals(betDaoInterface)
```
optional: we can remove one if statement and write:
if (field.isAnnotationPresent(Inject.class)
          && field.getType().equals(betDaoInterface)
          && betDaoImplClass.isAnnotationPresent(Dao.class))

Commit:
fix
